app/socket-apis/__tests__/upsert-rank.js

```diff
@@ -0,0 +1,103 @@
+import rankWhy from '../rank-why'
+import Ranks from '../../models/ranks'
+import { Mongo } from '@enciv/mongo-collections'
+import { MongoMemoryServer } from 'mongodb-memory-server'
+import { ObjectId } from 'mongodb'
+import upsertRank from '../upsert-rank'
+
+const USER1 = '6667d5a33da5d19ddc304a6b'
+const RANK1 = new ObjectId('6667d688b20d8e339ca50020')
+const RANK2 = new ObjectId('6667e4eea414d31b20dffb2f')
+let MemoryServer
+
+beforeAll(async () => {
+  MemoryServer = await MongoMemoryServer.create()
+  const uri = MemoryServer.getUri()
+  await Mongo.connect(uri)
+})
+
+beforeEach(async () => {
+  const collections = await Mongo.db.collections()
+  // Clear all collections
+  for (let collection of collections) {
+    await collection.deleteMany({})
+  }
+
+  jest.spyOn(console, 'error').mockImplementation(() => {})
+})
+
+afterEach(async () => {
+  console.error.mockRestore()
+})
+
+afterAll(async () => {
+  await Mongo.disconnect()
+  await MemoryServer.stop()
+})
+
+test('Insert a new rank document', async () => {
+  const rankObj = {
+    _id: RANK2.toString(),
+    parentId: 'parent-id-1',
+    round: 1,
+    stage: 'pre',
+    category: 'most',
+    discussionId: 'discussion-1',
```

Comment:
please make a const for this:
const discussion1='674e15d08074148b508f8a45'
It's passing as is right now, but future validation may cause this to fail.